File: chapitre-21/exemple-buffers.c
// ------------------------------------------------------------------
// exemple-buffers.c
// Fichier d'exemple du livre "Developpement Systeme sous Linux"
// (C) 2000-2019 - Christophe BLAESS <christophe@blaess.fr>
// https://www.blaess.fr/christophe/
// ------------------------------------------------------------------

#include <stdio.h>
#include <stdlib.h>

int main (void)
{
	char string[10];

	fprintf(stdout, "1 stdout : ligne + \\n\n");
	fprintf(stdout, "2 stdout : ligne seule");
	fprintf(stderr, "\n3 stderr : avant fflush(stdout)\n");
	fflush(stdout);
	fprintf(stderr, "\n4 stderr : apres fflush (stdout)\n");
	fprintf(stdout, "5 stdout : ligne seule ");
	fprintf(stderr, "\n6 stderr : avant fgets(stdin)\n");
	fgets(string, 10, stdin);
	fprintf(stderr, "\n7 stderr : apres fgets(stdin)\n");

	return EXIT_SUCCESS;
}

File: chapitre-21/exemple-buffers.py
#! /usr/bin/python
# ------------------------------------------------------------------
# exemple-buffers.py
# Fichier d'exemple du livre "Developpement Systeme sous Linux"
# (C) 2015-2019 - Christophe BLAESS <christophe@blaess.fr>
# https://www.blaess.fr/christophe/
# ------------------------------------------------------------------
from __future__ import print_function
import sys

print("1 stdout : ligne + \\n")
print("2 stdout : ligne seule", end='')
print("\n3 stderr : avant flush()", file=sys.stderr)
sys.stdout.flush()
print("\n4 stderr : apres flush ()", file=sys.stderr)
print("5 stdout : ligne seule ", end='')
print("\n6 stderr : avant sys.stdin.read()", file=sys.stderr)
sys.stdin.readline()
print("\n7 stderr : apres sys.stdin.read(stdin)", file=sys.stderr)

